tests/erica_legacy/elster_xml/grundsteuer/test_elster_grundstueck.py

```diff
@@ -0,0 +1,201 @@
+
+        assert result.E7411001 == 999999999999999
+
+
+class TestEEntwZust:
+    def test_if_bauerwartungsland_should_set_1(self):
+        result = EEntwZust("bauerwartungsland")
+
+        assert result.E7403051 == 1
+
+    def test_if_rohbauland_should_set_1(self):
+        result = EEntwZust("rohbauland")
+
+        assert result.E7403051 == 2
+
+
+class TestEAngGrund:
+    def test_if_valid_input_then_set_fields_correctly(self):
+        grundstueck = SampleGrundstueck().abweichende_enwticklung("rohbauland").parse()
+
+        result = EAngGrund(grundstueck)
+
+        assert result.Ang_Flaeche == EAngFlaeche(grundstueck)
+        assert result.Entw_Zust == EEntwZust(grundstueck.abweichende_entwicklung)
+        assert len(vars(result)) == 2
+
+    def test_if_no_abweichende_entwicklung_then_set_field_none(self):
+        grundstueck = SampleGrundstueck().parse()
+
+        result = EAngGrund(grundstueck)
+
+        assert result.Ang_Flaeche == EAngFlaeche(grundstueck)
+        assert result.Entw_Zust is None
+        assert len(vars(result)) == 2
+
+
+class TestEAngFlaeche:
+    def test_if_one_flurstueck_whole_should_calculate_correctly(self):
+        flurstueck1 = SampleFlurstueck().groesse(1000).w_einheit_zaehler("1.0000").w_einheit_nenner(1).parse()
+        grundstueck = SampleGrundstueck().bodenrichtwert("422,99").flurstuck(flurstueck1).parse()
+
+        result = EAngFlaeche(grundstueck)
+
+        assert result.E7403010 == 1000
+        assert result.E7403011 == "422,99"
+        assert len(vars(result)) == 2
+
+    def test_if_one_flurstueck_partial_should_calculate_correctly(self):
+        flurstueck1 = SampleFlurstueck().groesse(1000).w_einheit_zaehler("1.0000").w_einheit_nenner(2).parse()
+        grundstueck = SampleGrundstueck().flurstuck(flurstueck1).parse()
+
+        result = EAngFlaeche(grundstueck)
+
+        assert result.E7403010 == 500
+
+    def test_if_zaehler_with_nonzero_fraction_should_calculate_correctly(self):
+        flurstueck1 = SampleFlurstueck().groesse(1000).w_einheit_zaehler("1.2345").w_einheit_nenner(2).parse()
+        grundstueck = SampleGrundstueck().flurstuck(flurstueck1).parse()
+
+        result = EAngFlaeche(grundstueck)
+
+        assert result.E7403010 == 617
+
+    def test_if_zaehler_fraction_less_than_half_result_should_round_down_to_int(self):
+        flurstueck1 = SampleFlurstueck().groesse(1000).w_einheit_zaehler("1.0000").w_einheit_nenner(3).parse()
+        grundstueck = SampleGrundstueck().flurstuck(flurstueck1).parse()
+
+        result = EAngFlaeche(grundstueck)
+
+        assert result.E7403010 == 333
+
+    def test_if_zaehler_fraction_greater_than_half_result_should_round_down_to_int(self):
+        flurstueck1 = SampleFlurstueck().groesse(1000).w_einheit_zaehler("2.0000").w_einheit_nenner(3).parse()
+        grundstueck = SampleGrundstueck().flurstuck(flurstueck1).parse()
+
+        result = EAngFlaeche(grundstueck)
+
+        assert result.E7403010 == 666
```

Comment:
Should it round down? Is that said somewhere or did you just figure? Because I think if it's not specified, we should round as you normally would, not just cutting off the float part.

Reply:
That is specified by product - see https://steuerlotse.atlassian.net/browse/STL-2051